### PR TITLE
Implementation of collapsible sections

### LINKED ISSUE #163: Make each section (feats, actions, bonus, legendary, etc) an accordion that can be collapsed.
A new "mini-blocks" feature similar to compact-features but applied to each whole section heading.

The features section will be weird since it has no heading to act as a control.

### PR DESCRIPTION
**Pull request by jeremyregnerus**
Resolves #163 Make each section (feats, actions, bonus, legendary, etc) an accordion that can be collapsed.

- This implementation adds a show-collapsible option to enable or disable collapsible sections of the sheet.
   - I could not find a way to incorporate an `enabled` setting into the `collapsible` object which controls the state of the other sections, and have this option to enable / disable this from the module settings page.
   - This option can be controlled from the module settings or from the actor sheet
- Once enabled, the Features section will show a header (more about this below).
- Each header has a collapsible icon that appears when hovering over the header
   - This could be set to appear when hovering over the entire section, however I felt this could be distracting
- The collapsed state of each section is preserved when the actor sheet is closed and re-opened and across sessions
- If you disable the show-collapsible option, all of the sections _expand_
- This implementation utilizes a new header partial, which includes options for showing resources beyond legendary resources
   - This new header partial also contains partials which contain the section content, this is used to control the visibility of this section
   - This is done using a show / no show condition. I also considered adding the `hidden` class to the wrapper instead, but I'm not sure of the pros and cons of using either method
- Moved loot to the end of the character sheet. Currently it appears between legendary and lair actions
- Added google translated localization strings to localization files

https://user-images.githubusercontent.com/7407481/157364859-9e63d82e-ee87-47f3-ba2b-0461120baf1f.mp4

### CODE REVIEW

File: scripts/dnd5e/MonsterBlock5e.js [hunk 1]
```diff
@@ -637,7 +637,17 @@ export default class MonsterBlock5e extends ActorSheet5eNPC {
...
+
+		html.find(".collapsible").click((event) => {
+			event.preventDefault();
+
+			let control = event.currentTarget.dataset.control;
+			let collapsible = this.flagManager.flags.collapsible;
+
+			collapsible[control] = !collapsible[control];
+
+			this.flagManager.flags.collapsible = collapsible;
```
Comment (zeel01) [inline scripts/dnd5e/MonsterBlock5e.js:649]: `this.flags` is an alias to `this.flagManager.flags` that for some reason I haven't refactored all the code to use. It's defined up on line `49`.

File: templates/dnd5e/switches.hbs [hunk 2]
```diff
@@ -40,6 +40,12 @@
 		<li>
 			<a>{{ localize "MOBLOKS5E.DescriptionS" }}</a>
 			<ul>
+				{{> "modules/monsterblock/templates/dnd5e/parts/switch.hbs"
+					flag=flags.show-collapsible
+					control="show-collapsible"
+					on="MOBLOKS5E.show-collaspsible.disable"
```
Comment (zeel01) [inline templates/dnd5e/switches.hbs:46]: Typo here, "coll**asp**sible" on both strings, so it ends up not localizing.

**Review by zeel01 — CHANGES_REQUESTED** [review]
Alright, I made a lot of comments. Some of these things might be things I should work on, but I mentioned anyway for completions sake.

One thing I want to consider, and unfortunately it would be a lot of changes... `collapsible` is not an easy word to not typo, and every time I look at it I don't believe it's spelled that way even though Google tells me it.

I'm not sure if there is a great alternative. I usually refer to this sort of thing as an "accordion" but that's also kind of a tricky word to get right all the time. So idk if it makes sense to change this, or if there is a better term to use that will be less trouble.

Also, the changes in `main.hbs` are bit hard to follow in the way Github presents them, so I may have missed some things in that area,

File: monsterblock.css [hunk 3]
```diff
@@ -680,6 +680,14 @@ li:nth-last-of-type(2):after {
 	font-variant: small-caps;
 	border-bottom: 1px solid var(--heading-color);
 }
+.monsterblock .collapsible {
```
[the remark this answers is not in the document]
Comment (jeremyregnerus) [reply monsterblock.css:683]: Do you mean, you want two classes on the tag element, 1 for the display properties, and another for the command itself? Something like
```html
<a class="collapsible collapsible-button">
```

[the remark this answers is not in the document]
Comment (zeel01) [reply monsterblock.css:683]: No, I would just make it `<a class="collapsible-button">`. The idea being that the `<a>` tag is just a button, it's not itself collapsible. The `<section>` that gets hidden is "collapsible" but the anchor is just a button. So the class `.collapsible` is slightly confusing.

**Comment by jeremyregnerus** [comment]
Commit https://github.com/zeel01/MonsterBlocks/pull/169/commits/ff3f548bba5471051b43b50988b5828241f6b2fe
Right now, I have everything resolved with the exception of the float: right suggestion. I'm struggling to get a `flex` to work while keeping the Title left justified and the resources and collapsible switch right justified. Everything I do either adds a new line, or doesn't justify the last two.
Also, I did notice that `header-resource` is currently using `float: right` as well (which I've tried to remove as part of doing this). Possibly you could provide some insight on the correct way to do this.

**Comment by zeel01** [comment]
> Commit [ff3f548](https://github.com/zeel01/MonsterBlocks/commit/ff3f548bba5471051b43b50988b5828241f6b2fe) Right now, I have everything resolved with the exception of the float: right suggestion. I'm struggling to get a `flex` to work while keeping the Title left justified and the resources and collapsible switch right justified. Everything I do either adds a new line, or doesn't justify the last two. Also, I did notice that `header-resource` is currently using `float: right` as well (which I've tried to remove as part of doing this). Possibly you could provide some insight on the correct way to do this.

That's fine, I can fix that.

**Comment by jeremyregnerus** [comment]
This was my attempt at changing the floats to flex
https://github.com/zeel01/MonsterBlocks/pull/169/commits/4376c78a224272c10a29b23daefd66c63cf6bfa0

File: scripts/dnd5e/MonsterBlock5e.js [hunk 4]
```diff
@@ -99,12 +99,22 @@ export default class MonsterBlock5e extends ActorSheet5eNPC {
...
 			vttatokenizer: Boolean(window.Tokenizer)
 		}
+		data.info["hasAbilities"] = Boolean(
+			data.info.hasFeatures ||
+			data.info.hasActions ||
+			data.info.hasBonusActions ||
+			data.info.hasReactions ||
+			data.info.hasLegendaryActions ||
+			data.info.hasLair ||
+			data.info.hasLoot);
```
Comment (zeel01) [inline scripts/dnd5e/MonsterBlock5e.js:116]: Nitpick: I prefer to put the closing `);` on the next line.

File: templates/dnd5e/main.hbs [hunk 5]
```diff
@@ -1,93 +1,105 @@
...
-			{{> "modules/monsterblock/templates/dnd5e/parts/featureBlock.hbs" item=features.legResist.items.[0] actor=actor flags=flags}}
-		{{/if}}
-
-	{{! Actions }}
-		{{#if info.hasActions}}
-			<h2 class="section-header" >{{localize "DND5E.ActionPl"}}</h2>
-		{{/if}}
+{{! Features }}
+	{{#if info.hasFeatures}}
+		{{#> "modules/monsterblock/templates/dnd5e/collapsibleSection.hbs" title="DND5E.Features" hide-title=true section="feats" flag=flags.collapsed.feats }}
```
Comment (zeel01) [inline templates/dnd5e/main.hbs:12]: We could maybe use the [`lookup` helper](https://handlebarsjs.com/guide/builtin-helpers.html#lookup) to find the flag in `flags.collapsed` using the `section` parameter? In the partial it might look like:
```handlebars
{{#unless (and flags.show-collapsible (lookup flags.collapsed section))}}
    {{> @partial-block}}
{{/unless}}
```
Then we could drop the `flag` parameter.

**Review by zeel01 — CHANGES_REQUESTED** [review]

**Review by jeremyregnerus — COMMENTED** [review]
Updates made based on latest comments. Was trying to use lookup to simplify adding the attributes for value / max to the sectionHeader

**Review by zeel01 — APPROVED** [review]
Awesome. Nice work!
